src/Models/Market.php

```diff
@@ -35,11 +36,17 @@ protected static function boot()
         parent::boot();
 
         static::creating(function (Market $market) {
-            $market->slug = $market->slug ?: Str::slug($market->city);
-            $invalidSlugs = config('locations.invalid_slugs') ?? [];
-            if (in_array($market->slug, $invalidSlugs)) {
-                $market->slug = Str::slug("{$market->slug}-{$market->state}");
+            $market->updatingSlug();
+            $other_market = Market::where('slug', $market->slug)->first();
```

Comment:
Silently not creating the model when there is a conflict will likely cause errors or confusion in the future.  Should really throw an exception in my opinion.